libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { DefaultEntityStore } from '../store';
+import { EntityCollectionService } from '@ngrx/data';
+import { QueryBuilder } from '../query/query-builder';
+import { BaseEntity, BaseEntityDTO, BaseValue, RESTResource } from '@fundamental-ngx/store';
+import { Type } from '../../../domain/public_api'
+
+interface MoneyDTO {
+    amount: number;
+    currency: string
+}
+
+interface Item {
+    title: string;
+    price: number;
+    amount: Money;
+}
+
+interface ReqDTO extends BaseEntityDTO {
+    title: string;
+    amount: number;
+    lineItems: Array<LineItem>;
+}
+
+class Money extends BaseValue<MoneyDTO> {
+    constructor(value?: MoneyDTO) {
+        super(value);
+    }
+}
+
+@RESTResource({
+    root: 'http://www.example.com/v1/',
+    path: '/requisitions/:reqId/lineItems',
+})
+export class LineItem extends BaseValue<Item> {
+    amount: Money;
+    title: string;
+
+    constructor(dto?) {
+        super(dto);
+    }
+}
+
+interface ReqDTO extends BaseEntityDTO {
+    id: number;
+    title: string;
+    amount: number;
+    lineItems: Array<LineItem>;
+}
+
+class Requisition extends BaseEntity<ReqDTO> {
+    id: number;
```

Comment:
How we ended up with the discussion about the ID?  In Ariba I see we can have :

```
class Requisition {uniqueName: string}

class SourcingEvent {internalId: string;}

class SomeOtherEntity {id: string)

Now we need to have way to point to the correct key. 

What if our baseEntity will have method/getter/setter. 

identity(): string, so that in the Requisition about you will implement it and call the right DTO attribute? 

```
class Requisition extends BaseEntity {
   identity(): string {
    return  this.value.id;
   }
 
}
```

or maybe just getter/setter